misc/pytorch_toolkit/miriad_compression_codes/src/utils/inference_utils.py

```diff
@@ -0,0 +1,190 @@
+        self.bit_depth = bit_depth
+
+    def forward(self, x):
+        x = torch.round(x * (2**self.bit_depth - 1)).type(torch.int32)
+        return x
+
+# The Int->Float module
+class Int2Float(nn.Module):
+    def __init__(self, bit_depth=8):
+        super().__init__()
+        self.bit_depth = bit_depth
+
+    def forward(self, x):
+        x = x.type(torch.float32) / (2**self.bit_depth - 1)
+        return x
+
+def inference_model(config):
+
+    if config['phase'] == 1:
+        model = AutoEncoder(config['depth'], config['width'])
+    else:
+        model =  Decoder(config['depth'], config['width'])
+    
+    if config['bit_depth'] == 16:
+        float2int = Float2Int(12)
+        int2float = Int2Float(12)
+    else:
+        float2int = Float2Int(config['bit_depth'])
+        int2float = Int2Float(config['bit_depth'])
+    
+    # GPU transfer
+    if torch.cuda.is_available() and config['gpu']:
+        model = model.cuda()
+        model  = model.cuda()
+        float2int, int2float = float2int.cuda(), int2float.cuda()
+    
+    # load the given model
+    with open(os.path.abspath(config['model_file']), 'rb') as modelfile:
+        loaded_model_file = torch.load(modelfile)
+        model.load_state_dict(loaded_model_file['model_state'])
+        model.load_state_dict(loaded_model_file['model_state'])
+
+    images_transforms = torchvision.transforms.Compose(
+                                                [torchvision.transforms.Grayscale(),
+                                                torchvision.transforms.ToTensor()])
+    labels_transforms = torchvision.transforms.Compose(
+                                                [torchvision.transforms.Grayscale(),
+                                                torchvision.transforms.ToTensor()])
+    if not config['test_data_latentimage']:
+        infer_dataset = CustomDatasetPhase1(config['inferdata'],
+                                            transform_images=images_transforms,
+                                            transform_masks=labels_transforms,
+                                            preserve_names=True)
+        infer_dataloader = data.DataLoader(infer_dataset, batch_size=1,
+                                        num_workers=16, pin_memory=True,
+                                        shuffle=False)
+    else:
+        path_test_latent = config['inferdata']
+        path_test_gdtruth = config['inferdata']
+        infer_dataset = CustomDatasetPhase2(path_test_latent, path_test_gdtruth, transform_images=images_transforms, transform_masks=labels_transforms, preserve_name=True, mod =  0)
+        infer_dataloader = data.DataLoader(infer_dataset, batch_size=1, num_workers=16, pin_memory=True, shuffle=False)
+
+    model.eval()
+
+    with torch.no_grad():
+		# a global counter & accumulation variables
+        n = 0
+        all_bits, all_bpp, all_ssim, all_psnr = [], [], [], []
+        # avg_bits, avg_bpp, avg_ssim, avg_psnr = 0., 0., 0., 0.
```

Comment:
remove if these commented  variables are not used

Reply:
removed as they were not used and weren't affecting the program